extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java

```diff
@@ -143,6 +147,12 @@ public void transform(TransformationContext context) {
                 DefaultCommandListenersProvider.class,
                 MetricsCollectorFactory.class);
         additionalBean.produce(builder.build());
+
+        if (!capabilities.isCapabilityPresent(Capabilities.METRICS)) {
+            //disable fault tolerance metrics with the MP sys props and provides a No-op metric registry.
+            additionalBean.produce(new AdditionalBeanBuildItem(NoopMetricRegistry.class));
+            systemProperty.produce(new SystemPropertyBuildItem("MP_Fault_Tolerance_Metrics_Enabled", "false"));
```

Comment:
Is this property necessary? I don't see it being used anywhere

Reply:
This is a configuration option specified by MicroProfile Fault Tolerance to switch off metrics collection. SmallRye Fault Tolerance reads it.

Reply:
Got it, thanks!